### ISSUE TITLE
bug: Ion-Segment has layout error in Ion-Content with ion-padding

### ISSUE THREAD

**Issue by dythbj** [ionic-team/ionic-framework#29774]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/docs/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

Ion-Segment has layout error in Ion-Content with ion-padding, it's nomarl on web and Simulator, but has layout error on device like below shows.

eg1: In Ion-Content with ion-padding, covered by Ion-Content on the left and right
![9786578627148CE900EF099DAE013CA3](https://github.com/user-attachments/assets/7ca18135-ba42-4203-969b-dd1a4dc8a2f7)
![2457A522FF5E08A3D6C0C14B94FAD054](https://github.com/user-attachments/assets/29e3c9a5-2389-4045-907b-a8c5489b7c37)

### Expected Behavior

Ion-Segment has right layout on device like this:
![localhost_8100_tutorial(Pixel 7)](https://github.com/user-attachments/assets/eb8ae73e-e10a-4189-ba4d-4a1408dc08b1)

### Steps to Reproduce

```
<ion-header [translucent]="true">
  <ion-toolbar>
    <ion-title>signup</ion-title>
    <ion-segment value="default">
      <ion-segment-button value="default">
        <ion-label>默认</ion-label>
      </ion-segment-button>
      <ion-segment-button value="terrain">
        <ion-label>地形</ion-label>
      </ion-segment-button>
      <ion-segment-button value="satellite">
        <ion-label>影像</ion-label>
      </ion-segment-button>
      <ion-segment-button value="3d">
        <ion-label>3D</ion-label>
      </ion-segment-button>
    </ion-segment>
  </ion-toolbar>
</ion-header>

<ion-content [fullscreen]="true" class="ion-padding">
  <ion-header collapse="condense">
    <ion-toolbar>
      <ion-title size="large">signup</ion-title>
    </ion-toolbar>
  </ion-header>
  <ion-searchbar search-icon="./../../assets/img/search.svg" color="light" animated="true" placeholder="搜索地名、地址"></ion-searchbar>
  <ion-label class="ion-padding-horizontal"><h2>地图显示</h2></ion-label>
  <ion-segment value="default">
    <ion-segment-button value="default">
      <ion-label>默认</ion-label>
    </ion-segment-button>
    <ion-segment-button value="terrain">
      <ion-label>地形</ion-label>
    </ion-segment-button>
    <ion-segment-button value="satellite">
      <ion-label>影像</ion-label>
    </ion-segment-button>
    <ion-segment-button value="3d">
      <ion-label>3D</ion-label>
    </ion-segment-button>
  </ion-segment>
  <ion-label class="ion-padding-horizontal"><h2>地图显示</h2></ion-label>
</ion-content>
```

### Code Reproduction URL

https://github.com/dythbj/segment/blob/main/src/app/tab1/tab1.page.html

### Ionic Info

Ionic:

   Ionic CLI                     : 7.2.0 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 8.2.0
   @angular-devkit/build-angular : 17.3.8
   @angular-devkit/schematics    : 17.3.8
   @angular/cli                  : 17.3.8
   @ionic/angular-toolkit        : 11.0.1

Capacitor:

   Capacitor CLI      : 6.0.0
   @capacitor/android : 6.0.0
   @capacitor/core    : 6.0.0
   @capacitor/ios     : 6.0.0

Utility:

   cordova-res : not installed globally
   native-run  : 2.0.1

System:

   NodeJS : v20.14.0 (/usr/local/bin/node)
   npm    : 10.7.0
   OS     : macOS Sonoma 14.5

### Additional Information

It's nomarl on web and Simulator, but has error on device, so i cann't give Reproduction Url.

**Comment by dev-charles15531** [comment]
I don't seem to understand where the error is. Everything looks okay to me.


Are you trying to make the segment span full width like the segment in the header?

**Comment by dythbj** [comment]
> I don't seem to understand where the error is. Everything looks okay to me.
> 
> Are you trying to make the segment span full width like the segment in the header?

Nope, It's same situation when i put the Ion-Segment in Ion-card-content

![image](https://github.com/user-attachments/assets/dedad64b-cbc2-4bd5-b3ff-2ba53b77b3d9)

as you can see, left segment button's  indicator missing a part(also on the right segment button)

![image](https://github.com/user-attachments/assets/a6435dfc-2724-4314-a184-a5a6d03360d6)

![image](https://github.com/user-attachments/assets/2f01039e-85c4-4e25-830a-aa95b1d56c35)

Just have a test like this:
<ion-card>
          <ion-card-content>
            <ion-segment value="default">
              <ion-segment-button value="default">
                <ion-img src="./../../assets/img/ic_default.png"></ion-img>
                <ion-label>默认</ion-label>
              </ion-segment-button>
              <ion-segment-button value="terrain">
                <ion-img src="./../../assets/img/ic_terrain.png"></ion-img>
                <ion-label>地形</ion-label>
              </ion-segment-button>
              <ion-segment-button value="satellite">
                <ion-img src="./../../assets/img/ic_satellite.png"></ion-img>
                <ion-label>影像</ion-label>
              </ion-segment-button>
              <ion-segment-button value="3d">
                <ion-img src="./../../assets/img/ic_3d.png"></ion-img>
                <ion-label>3D</ion-label>
              </ion-segment-button>
            </ion-segment>
          </ion-card-content>
        </ion-card>

**Comment by dev-charles15531** [comment]
If you don't need the padding added, you can always remove the `ion-card-content`.
The `ion-card-content` component introduces a new padding in the container as this is only recommended when you need a text content.

```
<ion-card>
  <!-- <ion-card-content> -->
    <ion-segment value="default">
      <ion-segment-button value="default">
        <ion-label>Default</ion-label>
      </ion-segment-button>
      <ion-segment-button value="segment">
        <ion-label>Segment</ion-label>
      </ion-segment-button>
      <ion-segment-button value="aa">
        <ion-label>aaDefault</ion-label>
      </ion-segment-button>
      <ion-segment-button value="bb">
        <ion-label>bbSegment</ion-label>
      </ion-segment-button>
    </ion-segment>
  <!-- </ion-card-content> -->
</ion-card>
``` 

![crispy-acorn-7wwxxqvg4grfxj7g-8080 app github dev_cus-test(iPhone 14 Pro Max)](https://github.com/user-attachments/assets/6c1a2072-5003-443f-ae94-d81688117bca)

**Comment by dythbj** [comment]
> If you don't need the padding added, you can always remove the `ion-card-content`. The `ion-card-content` component introduces a new padding in the container as this is only recommended when you need a text content.
> 
> ```
> <ion-card>
>   
>     <ion-segment value="default">
>       <ion-segment-button value="default">
>         <ion-label>Default</ion-label>
>       </ion-segment-button>
>       <ion-segment-button value="segment">
>         <ion-label>Segment</ion-label>
>       </ion-segment-button>
>       <ion-segment-button value="aa">
>         <ion-label>aaDefault</ion-label>
>       </ion-segment-button>
>       <ion-segment-button value="bb">
>         <ion-label>bbSegment</ion-label>
>       </ion-segment-button>
>     </ion-segment>
>   
> </ion-card>
> ```
> 
> ![crispy-acorn-7wwxxqvg4grfxj7g-8080 app github dev_cus-test(iPhone 14 Pro Max)](https://private-user-images.githubusercontent.com/68998548/358329648-6c1a2072-5003-443f-ae94-d81688117bca.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwMzQzMzEsIm5iZiI6MTcyNDAzNDAzMSwicGF0aCI6Ii82ODk5ODU0OC8zNTgzMjk2NDgtNmMxYTIwNzItNTAwMy00NDNmLWFlOTQtZDgxNjg4MTE3YmNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE5VDAyMjAzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2MmM2NTE4MTY1NzllOWU4NDY3NDIxYjM5MTg3ZWJkZWE2ZWZlYTJkODQ5YjIwODI0YTllNzM2YjZlYWM5N2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qh7f4kYqvdI9LJlLe4r8TZi2nz9dPvxFYijTBhAak2w)

Yes, @dev-charles15531 , this can make it display nomarl, but it's not my point.
I want the Ion-segemnt follow the parent container's layout.
Eg: A Ion-content with ion-padding setted，also I want to emphasize that it's looks normal on web & Simulator, but has layout error on device.

**Comment by thetaPC** [comment]
Thank you submitting the issue!

Verified. This reproduces on latest, and in Chrome, Safari and Firefox alike.

It isn't a device bug, which is why it looked inconsistent. Segment buttons have a fixed minimum width of 90px on Material Design (70px on iOS), so four buttons need at least 360px. When the segment's container is narrower than that, as it is once ion-padding takes 32px out of the content width, the buttons no longer fit and the segment centers the overflow. That splits the excess evenly across both edges and clips it, which is why part of the first button and its indicator disappear on the left and the last button is cut on the right. Your simulator and browser window were simply wider than your phone, so they stayed above the threshold.

A workaround you can use today, letting the labels shrink and truncate instead:

```css
ion-segment-button {
  min-width: 0;
}
```

If you'd rather keep the labels at full length, set `scrollable="true"` on the `ion-segment` and it will scroll horizontally instead of clipping.

Two notes on suggestions above. Removing `ion-card-content` only works because it widens the container back past the threshold; it will break again on a narrower device, and it doesn't help the `ion-content class="ion-padding"` case you originally asked about. And `min-width: auto`, which we recommended on an older issue, resolves to the label's own intrinsic width, so it can end up wider than 90px and make the clipping worse. `min-width: 0` is the reliable one.
